openmdao/docs/usr-guide/tutorials/beam-sizing.rst

```diff
@@ -0,0 +1,487 @@
+
+George knows that the best way to meet his requirements will be to choose a steel wide flange beam.  George called his local steel retailer and found that the largest, heaviest, and strongest 8 inch beam they sell is a W8x58 beam, meaning that it is about 8 inches high and weighs 58 pounds per foot length.  
+
+.. figure:: basement_actual.png
+   :align: center
+   :alt: An actual W8x58 steel beam supporting the floor joists in the basement.
+
+   An actual W8x58 steel beam supporting the floor joists in the basement.
+
+
+
+Objective
+-----------------
+Maximize room addition square footage.  In other words, find the optimum length and width of the room addition while satisfying the constraints.  For this exercise, all calculations will be done in inches and pounds.
+
+.. figure:: basement_top_view.png
+   :align: center
+   :alt: Top view sketch of room addition basement.
+
+   Top view sketch of room addition basement.
+
+Constraints
+---------------------
+- Use a W8x58 wide flange beam made from ASTM A992 steel.
+- Beam will only be supported at the two ends.
+- Achieve a deflection rating of at least L/720.
+- Make sure beam safely satisfies bending stress requirements.
+- Make sure beam safely satisfies shear stress requirements.
+- Room length is greater than room width.
+
+Constants
+---------------
+The constants used in this tutorial are:
+
+.. testcode:: Beam
+    
+    E = 29000000 #modulus of elasticity (constant 29000000psi for ASTM A992 Grade 50 steel) 
+    I = 228 #Ix = moment of Inertia (constant 228in4 for the W8x58 beam) 
+    BEAM_WEIGHT_LBS_PER_IN = 58.0 / 12.0 #self weight of beam per unit length (58 lbs/ft or 4.83 lbs/in.)
+    DEAD_LOAD_PSI = 20.0 / 144 #The dead load is 20psf or 0.1389psi.
+    LIVE_LOAD_PSI = 50.0 / 144 #The live load is 50psf or 0.3472psi.
+    TOTAL_LOAD_PSI = DEAD_LOAD_PSI + LIVE_LOAD_PSI #total load
+    BEAM_HEIGHT_IN = 8.75 #inches
+    YIELD_STRENGTH_PSI = 50000 #The maximum yield strength Fy for ASTM A992 Grade 50 steel is 50,000 psi
+    CROSS_SECTIONAL_AREA_SQIN = 17.1 #sq in
+
+Room Area Discipline
+----------------------
+We want to maximize room area.  Room area is given by the following equation.
+
+.. math:: 
+    \mathrm{room\_area} = \mathrm{room\_length} * \mathrm{room\_width}    
+
+However, in OpenMDAO, problems must be written as a minimization problem.  The best way to do that is to negate the equation.  Therefore, we want to minimize neg_room_area such that
+
+.. math:: 
+    \mathrm{neg\_room\_area} = -(\mathrm{room\_length} * \mathrm{room\_width})
+    :label: neg_room_area
+
+Now we can find our derivatives:
+
+.. math:: 
+    \frac{d \mathrm{neg\_room\_area}} {d \mathrm{room\_width}} = -\mathrm{room\_length}
+           
+    \frac{d \mathrm{neg\_room\_area}} {d \mathrm{room\_length}} = -\mathrm{room\_width}
+
+Now we can take this equation and create a `Component` called `NegativeAreaDiscipline`.
+
+.. testcode:: Beam
+
+    class NegativeAreaDiscipline(Component):    
```

Comment:
I suggest renaming all of these 'disciplines' to components, e.g., change NegativeAreaDiscipline to NegativeAreaComp. These aren't really different disciplines.